### PR TITLE
CAPI-1.7.4: Move core-CAPI dependencies to v1.7.4.

### PR DESCRIPTION
**Pull request by arunmk**
## Description
Please provide a brief description of the changes proposed in this Pull Request
- Imported core CAPI 1.7.4 and made changes needed since some of the methods changed. This upgrade also removed v1alpha4 compatibility since core-capi removed it fro v1.7.0 onwards.

## Checklist
- [X] tested locally
- [X] updated any relevant dependencies
- [X] updated any relevant documentation or examples

## API Changes
Are there API changes?
- [ ] Yes
- [X] No

If yes, please fill in the below

1. Updated conversions?
    - [ ] Yes
    - [ ] No
    - [ ] N/A
2. Updated CRDs?
    - [ ] Yes
    - [ ] No
    - [ ] N/A
3. Updated infrastructure-components.yaml?
    - [ ] Yes
    - [ ] No
    - [ ] N/A
4. Updated `./examples/capi-quickstart.yaml`?
    - [ ] Yes
    - [ ] No
    - [ ] N/A
5. Updated necessary files under `./infrastructure-vcd/v1.0.0/`?
   - [ ] Yes
   - [ ] No
   - [ ] N/A

## Issue
If applicable, please reference the relevant issue

Fixes #

- - -
This change is [Reviewable](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/659)

### CODE REVIEW

**Review by lzichong — APPROVED** [review]
LGTM, I think we can remove the .DS_STORE file in controllers

File: main.go [hunk 1]
```diff
@@ -41,18 +45,22 @@ import (
 var (
 	myscheme = runtime.NewScheme()
 	setupLog = ctrl.Log.WithName("setup")
+
+	metricsAddr          string
```
Comment (arunmk) [inline main.go:49]: This variable can be removed

**Review by sakthisunda — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/659)* status: 0 of 467 files reviewed, 2 unresolved discussions (waiting on @arunmk, @rocknes, and @sahithi)

*[a discussion](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/659#-O8NOvOeBBJEGfmBKPwN:-O8NOvOeBBJEGfmBKPwO:bh4kc8b) (no related file):*
should folder api/v1alpha4 also be removed ?

___

**Review by sakthisunda — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/659)* status: 0 of 467 files reviewed, 1 unresolved discussion (waiting on @rocknes and @sahithi)

**Comment by arunmk** [comment]
> > _[Reviewable](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/659)_ status: 0 of 467 files reviewed, 2 unresolved discussions (waiting on @arunmk, @rocknes, and @sahithi)
> 
> _[a discussion](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/659#-O8NOvOeBBJEGfmBKPwN:-O8NOvOeBBJEGfmBKPwO:bh4kc8b) (no related file):_ should folder api/v1alpha4 also be removed ?

@sakthisunda I have removed it but github does not show it well. For verification, please look at the branch itself: https://github.com/arunmk/cluster-api-provider-cloud-director/tree/ak/capi-1.7.4/api

**Review by lzichong — APPROVED** [review]
Updated changes LGTM

**Review by rocknes — APPROVED** [review]
LGTM

**Comment by sakthisunda** [comment]
@arunmk Should there also be new files templates/cluster-template-crs.yaml and templates/cluster-template.yaml  for TKG 2.5.2 supported kubernetes versions ?

File: templates/infrastructure-components.yaml
Comment (sahithi) [inline templates/infrastructure-components.yaml:1]: nit: CRDs seem to have v1alpha4 labels. We can remove them for the correctness sake.

**Review by sahithi — COMMENTED** [review]
Please update metadata.yaml to remove v1alpha4

**Pull request merged**